languages/swift/exercises/concept/structs-and-classes/.docs/introduction.md

```diff
@@ -0,0 +1,158 @@
+
+They each provide some additional functionality, which is out of scope for this exercise.
+
+### Defining structs and classes
+
+Both structs and classes are defined in roughly the same way. They start with the appropriate keyword followed by the type name that is being defined, then the body of the struct/class follows, placed between curly braces.
+
+The body may consist of stored properties, which are defined and behave just like regular constants or variables.
+
+```swift
+struct CharacterStats {
+  var health = 0.0
+  var speed = 0
+  var strength = 0
+}
+
+class GameCharacter {
+  var stats = CharacterStats()
+  var characterClass: String?
+  var name: String?
+  var active = false
+  let id = makeRandomID()
+}
+```
+
+Here we have a struct that contains the statistics that will be associated with a character in a game and a struct that contains information about the character. Note that the `Character` class contains an instance of the `CharacterStats` struct inside of it.
+
+### Instances
+
+As noted above, defining a struct or class is just defining a new _type_. It is just the blueprint for what the values of that type will look like, but it does not actually create any values of that type for you to work with.
+
+In order to create an _instance_ of that type, you need to write the name of the type followed by a pair of parentheses.
+
+```swift
+let someStats = CharacterStats()
+let someCharacter = GameCharacter()
+```
+
+This will create values of these types, where the properties are populated with the default values supplied in the definition. Note that in optional cases like GameCharacter's `name` property, unless a value is provided, the property will default to nil, just like defining regular optional types where a value is not immediately provided.
+
+With structs, Swift automatically provides something called a memberwise initializer, where values for the structs properties may be provided inside the parentheses which will override the default values in the definition.
+
+```swift
+let differentStats = CharacterStats(health: 100.0, speed: 6, strength: 18)
+```
+
+### Accessing properties
+
+Struct and class properties can be accessed using _dot notation_ where the name of the value is followed by a dot (`.`) and the name of the property. If a property of a struct or class has properties of its own, this dot notation can be used to access these nested properties as well.
+
+This notation can be used both to retrieve the property's value and, where allowed, to change it.
+
+```swift
+someStats.health
+// => 0
+differentStats.health
+// => 100
+someCharacter.name
+// => nil
+
+someStats.health = 87.3
+someStats.health
+// => 87.3
+someCharacter.name = "Luther"
+someCharacter.name
+// => "Luther"
+someCharacter.stats = differentStats
+someCharacter.stats.health
+// => 100
+someCharacter.id  = "new id"
+// Error: Cannot assign to property: 'id' is a 'let' constant
```

Comment:
Maybe shorten this a bit?

Reply:
Shortened in the introduction, left as is in the after. Same with next comment.